udata/harvest/backends/dcat.py

```diff
@@ -87,16 +94,18 @@ def parse_graph(self, url, fmt):
 
         return graph
 
+    def get_node_from_item(self, item):
+        if 'nid' in item.kwargs and 'type' in item.kwargs:
+            nid = item.kwargs['nid']
+            return URIRef(nid) if item.kwargs['type'] == 'uriref' else BNode(nid)
+
     def process(self, item):
         graph = Graph(namespace_manager=namespace_manager)
-        data = item.kwargs.get('graph', self.job.data['graph'])  # handles legacy graphs
-        node = None
-
-        graph.parse(data=bytes(data, encoding='utf8'), format='json-ld')
+        data = self.job.data['graph']
+        format = self.job.data['format']
 
-        if 'nid' in item.kwargs and 'type' in item.kwargs:
-            nid = item.kwargs['nid']
-            node = URIRef(nid) if item.kwargs['type'] == 'uriref' else BNode(nid)
+        node = self.get_node_from_item(item)
+        graph.parse(data=bytes(data, encoding='utf8'), format=format)
 
         dataset = self.get_dataset(item.remote_id)
         dataset = dataset_from_rdf(graph, dataset, node=node)
```

Comment:
Sadly, this line fails if mongo is not started due to https://github.com/opendatateam/udata/blob/d626a2bc910639d4bf885f4ce37f3013f0852444/udata/core/dataset/rdf.py#L420

I guess it would be better if it doesn't fail on this ?

Reply:
Do we have a valid use case where Mongo is not running (except when you forget to start it locally 😅)?

Agreed it would be better, not sure it's worth the trouble.

Reply:
:laughing: Nope, probably not worth the trouble!